app_dart/lib/src/request_handlers/get_timeseries_history.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/model/appengine/benchmark_data.dart';
+import 'package:cocoon_service/src/model/appengine/time_series_value.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../model/appengine/commit.dart';
+import '../model/appengine/key_helper.dart';
+import '../model/appengine/time_series.dart';
+import '../model/appengine/time_series_entity.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../request_handling/regular_request_handler.dart';
+import '../service/datastore.dart';
+
+@immutable
+class GetTimeSeriesHistory extends RegularRequestHandler<GetTimeSeriesHistoryResponse> {
+  const GetTimeSeriesHistory(
+    Config config, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+  })  : datastoreProvider =
+            datastoreProvider ?? DatastoreService.defaultProvider,
+        super(config: config);
+
+  final DatastoreServiceProvider datastoreProvider;
+  static const String timeSeriesKeyParam = 'TimeSeriesKey';
+  static const String startFromParam = 'StartFrom';
+
+  @override
+  Future<GetTimeSeriesHistoryResponse> post() async {
+    checkRequiredParameters(<String>[timeSeriesKeyParam]);
+    const int maxRecords = 1500;
+    final DatastoreService datastore = datastoreProvider();
+    final KeyHelper keyHelper = KeyHelper(applicationContext: AppEngineContext(false,'','','','','',Uri()));
```

Comment:
I don't think we should be passing in the `applicationContext` like this here.  It seems like this was done for tests - we should just make the test be able to pass in an application context to the ctor, like we're doing with `datastoreProvider`.

Reply:
The different part from other existing APIs is: 
for this API: we do not want to have authentication, namely no login needed to view on the benchmark page
for other APIs: we need authenticated client to move forward.

So, not only for test, even in the code, we do not need any authenticated application context. That's why I simply put it like this.

Reply:
Doesn't applicationContext handle more than just authentication though?

Reply:
Existing other APIs are quoting keyHelper like below:

    final ClientContext clientContext = authContext.clientContext;
    final KeyHelper keyHelper =
        KeyHelper(applicationContext: clientContext.applicationContext);

authContext is defined in api_request_hander.dart

    AuthenticatedContext get authContext => getValue<AuthenticatedContext>(ApiKey.authContext);
    context = await authenticationProvider.authenticate(request);

Then it goes to the authentication.authenticate(request).
Since this API is public, we do not pass authProvider when called.. What we need here is simply an empty AppEngineContext to move forward.